packages/react/src/components/TextArea/TextArea.tsx

```diff
@@ -155,22 +161,67 @@ const TextArea = React.forwardRef((props: TextAreaProps, forwardRef) => {
   const prefix = usePrefix();
   const { isFluid } = useContext(FormContext);
   const { defaultValue, value, disabled } = other;
+
   const [textCount, setTextCount] = useState(
-    defaultValue?.toString().length || value?.toString().length || 0
+    defaultValue
+      ? counterMode === 'character'
+        ? defaultValue.toString().length
+        : defaultValue.toString().match(/\w+/g)?.length || 0
+      : value
+      ? counterMode === 'character'
+        ? value.toString().length
+        : value.toString().match(/\w+/g)?.length || 0
+      : 0
```

Comment:
this is really hard to read, could we offload this logic onto a `getInitialTextCount()` function and use if statements instead?

Reply:
done